ganttproject/src/net/sourceforge/ganttproject/DownloadWorker.java

```diff
@@ -0,0 +1,150 @@
+/*
+Copyright 2018 Oleksii Lapinskyi, BarD Software s.r.o
+
+This file is part of GanttProject, an opensource project management tool.
+
+GanttProject is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+GanttProject is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with GanttProject.  If not, see <http://www.gnu.org/licenses/>.
+*/
+package net.sourceforge.ganttproject;
+
+import net.sourceforge.ganttproject.gui.UIFacade;
+import net.sourceforge.ganttproject.gui.update.RestartDialog;
+
+import javax.swing.*;
+import java.io.*;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.List;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+
+public class DownloadWorker extends SwingWorker<Void, Integer> {
+  private static final int BUFFER_SIZE = 4096;
+  private static final String PLUGINS_FOLDER = "/plugins";
+  private String myDownloadURL;
+  private UIFacade myUiFacade;
+
+  public DownloadWorker(UIFacade uiFacade, String downloadURL) {
+    myUiFacade = uiFacade;
+    myDownloadURL = downloadURL;
+  }
+
+  @Override
+  protected void process(List<Integer> chunks) {
+    myUiFacade.setDownloadProgress(chunks.get(chunks.size() - 1));
```

Comment:
I suggest using UIFacade.setStatusText for this purpose (construct the string to be shown right here). Currently that method has no implementation, but it is trivial to implement.

Reply:
UIFacade.setStatusText is already used from other methods. And if I'll implement this method - it may show also other information in status bar. I think that it may create some issues with unexpected information on status bar..

Reply:
Thanks for pointing this out. It appears that most of those usages are pointless, I'll remove them.

Reply:
In this case I'm able to use UIFacade.setStatusText  safely?